knative-operator/pkg/common/metrics.go

```diff
@@ -6,21 +6,23 @@ import (
 )
 
 var (
-	KnativeServingReadyG = prometheus.NewGauge(
+	KnativeServingUpG = prometheus.NewGauge(
 		prometheus.GaugeOpts{
-			Name: "knative_serving_ready",
-			Help: "Reports if Knative Serving is up",
+			Name:        "knative_up",
+			Help:        "Reports if a Knative component is up",
+			ConstLabels: map[string]string{"type": "serving_status"},
 		},
 	)
-	KnativeEventingReadyG = prometheus.NewGauge(
+	KnativeEventingUpG = prometheus.NewGauge(
 		prometheus.GaugeOpts{
-			Name: "knative_eventing_ready",
-			Help: "Reports if Knative Eventing is up",
+			Name:        "knative_up",
+			Help:        "Reports if a Knative component is up",
```

Comment:
Help msg needs to be the same otherwise things fail.

Reply:
Rather than declaring 2 gauge metrics, you can use a `prometheus.GaugeVec`.

```
knativeUp = prometheus.NewGaugeVec(prometheus.GaugeOpts{...}, []string{"type"})

KnativeEventingUpG = knativeUp.WithLabelValues("eventing_status")
KnativeServingUpG = knativeUp.WithLabelValues("serving_status")
```